src/snap.py

```diff
@@ -263,3 +263,31 @@ def __str__(self):
     def __repr__(self):
         return '%s:%s' % (self.__class__.__name__, self.name)
 
+    @staticmethod
+    def packetise_snapdata(data, eof_key='eof', packet_length=-1):
+        """
+        Use the given EOF key to packetise a dictionary of snap data
+        :param data: a dictionary containing snap block data
+        :param eof_key: the key used to identify the packet boundaries
+        :param packet_length: check the length of the packets against
+        this as they are created (in 64-bit words)
+        :return: a list of packets
+        """
+        class PacketLengthError(Exception):
+            pass
+        _current_packet = {}
```

Comment:
Not really necessary to use _private vars here since they are not exposed anywhere outside this method?